aikau/src/main/resources/alfresco/layout/UploadContainer.js

```diff
@@ -100,18 +104,40 @@ define(["dojo/_base/declare",
        * @instance
        */
       setDndHighlightDimensions: function alfresco_documentlibrary__AlfDndDocumentUploadMixin__setDndHighlightDimensions() {
-         if (this.proxyDragAndDropNode && query(this.proxyDragAndDropNode).length === 1)
+
+         // proxyDragAndDropNode can be an array of strings or a comma-separated list
+         if (this.proxyDragAndDropNode && !(this.proxyDragAndDropNode instanceof Array))
```

Comment:
`Array.isArray()` is available from IE9. An addition to the `shims.js` file would probably be the most suitable way to fix this lack of support in IE8, as it will be temporarily needed and in a single place to remove easily.

The `ObjectTypeUtils.isArray()` method is merely a wrapper for `dojo/_base/lang.isArray()` which has the following test in it:

```javascript
return it && (it instanceof Array || typeof it == "array");
```

In fact, in their comments for the function, Dojo even have written "Does not work on Arrays created in other windows."

If you need any help updating `shims.js` (although it should be obvious) then please holler. We can then use `Array.isArray()` without worrying about the very unlikely event of handling an array across frames.